### ISSUE TITLE
Doesn't work when components are with "render" as a property in class

### ISSUE THREAD

**Issue by ezhlobo** [gajus/react-css-modules#204]
Hello.

Here is an example of my component:

```js
class Lists extends React.Component {
  render = () => (
    <div styleName="test">Lists</div>
  )
}

export default CSSModules(Lists, { test: "test-1" });
```

But it throw error: 
>Uncaught TypeError: Cannot call a class as a function

I forked your repository to send a PR with failed test case but I can't add a commit while tests are failed :).

Well. I can fix it but let me know do you actually want to support such methods defined as property.

**Comment by gajus** [comment]
Because of this and similar issues, I strongly encourage to evaluate whether https://github.com/gajus/babel-plugin-react-css-modules is not a feasible alternative.

**Comment by ezhlobo** [comment]
@gajus thank you for a quick answer.

I took a look at that project first, I found a link to it in Readme. But actually it's hard for me to understand how to use it with [Create React App](https://github.com/facebookincubator/create-react-app).

Could you point me to expanded guide how to use `babel-plugin-react-css-modules`? It would be helpful but I guess it isn't possible to connect with Create React App.

Also I'm afraid about so big dependency list in `babel-plugin-react-css-modules`.

**Comment by gajus** [comment]
> Also I'm afraid about so big dependency list in babel-plugin-react-css-modules.

Has 0 dependencies.

**Comment by ezhlobo** [comment]
@gajus yeah. Could you also add a good description how to use that plugin?

And if you are not going to add support class' properties for this project, just close this issue ;).

**Comment by gajus** [comment]
> And if you are not going to add support class' properties for this project, just close this issue ;).

I am not sure what you are trying to do here. Can you refer to the React documentation?

Don't you mean:

```js
class Lists extends React.Component {
  render () (
    <div styleName="test">Lists</div>
  )
}
```

Whats the reason for using arrow function?

**Comment by ezhlobo** [comment]
@gajus it's kind of styleguide.

We started use properties in order not to bind methods to current instance. But then we noticed that render is much shorter with arrow function than as method. So we decided to use class' properties for all non-lifecycle methods and for `render`. Here is an example how real component can look:

```js
class Component extends React.Component {
  render = () => (
    <button type="button" onClick={this.handleClick}>Click me!</button>
  )

  method = () =>
    'Hello'

  handlerClick = () => 
    alert(this.method())
}
```

Click on button will alert "Hello".

**Comment by gajus** [comment]
I am pretty sure that the reason this works is a bug in one of your transpilers.

Which TC proposal enables this syntax?

**Comment by ezhlobo** [comment]
Sorry, I'm not familiar with TC but in Create React App it's handled by `babel-plugin-transform-class-properties`. I know that this project supports such syntax too.

And the problem is in your method `isReactComponent` and in `extendReactComponent`.

Well, let me fix it and create PR so we will be able to discuss it more productively. Today I have day off so I'm not sure when I'll be able to do it but I'll try as soon as possible.

**Comment by gajus** [comment]
Sorry, for not being more clear. I meant https://github.com/tc39.

In your case, thats https://github.com/tc39/proposal-class-public-fields.

Have you got the appropriate settings configured?

https://github.com/gajus/react-css-modules#options

**Comment by diegodoumecq** [comment]
I stumbled into this bug myself and the error displayed sent me off on a wild-goose chase. I am not familiar with the codebase but it shouldn't be that hard to add the ability to use render as a property, right?

In any case, if it turns out to be impossible to fix, a more appropriate error message and a fair warning on the readme would be VERY helpful.

**Comment by ezhlobo** [comment]
@gajus thanks for explanation. Yeah, I took a look at options. Can you please check my PR (#210), I added fix and tests. BTW don't you want to remove running tests as a git-prehook because it's impossible to add failed tests.

@diegodoumecq thanks for a sign in this issue. I sent a PR with fix so it would be great if you test it on your use case.

**Comment by FezVrasta** [comment]
This happens even if you use `render` with the "classic" method, see https://github.com/gajus/react-css-modules/issues/196

**Comment by ezhlobo** [comment]
@FezVrasta the issue you pointed is about different problem. Isn't it?

**Comment by FezVrasta** [comment]
Yes sorry they seemed related at first glance
